src/main/java/com/greenfox/tribesoflagopus/backend/service/TokenService.java

```diff
@@ -0,0 +1,72 @@
+package com.greenfox.tribesoflagopus.backend.service;
+
+import com.greenfox.tribesoflagopus.backend.model.entity.User;
+import com.greenfox.tribesoflagopus.backend.repository.UserRepository;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.impl.crypto.MacProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.HashMap;
+
+@Service
+public class TokenService {
+
+  @Autowired
+  UserRepository userRepository;
+
+  private Key key;
+
+  public void saveTokenToUser(User user){
+    userRepository.findByUsername(user.getUsername()).setToken(generateToken(user));
+    userRepository.save(user);
```

Comment:
You forgot to save your user object that you just fetched from the db in line 24.
You should save that object back to the db in line 25.

Or it's even better if you would have a function in your repository that adds a given token to an existing user.
That would cause half times DB queries.